src/plugins/dist_matrixops/dist_identity.cpp

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2020 Nanmiao Wu
+// Copyright (c) 2020 Bita Hasheminezhad
+// Copyright (c) 2017-2020 Hartmut Kaiser
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/execution_tree/annotation.hpp>
+#include <phylanx/execution_tree/localities_annotation.hpp>
+#include <phylanx/execution_tree/locality_annotation.hpp>
+#include <phylanx/execution_tree/meta_annotation.hpp>
+#include <phylanx/execution_tree/primitives/node_data_helpers.hpp>
+#include <phylanx/execution_tree/tiling_annotations.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/plugins/dist_matrixops/dist_identity.hpp>
+#include <phylanx/plugins/dist_matrixops/tile_calculation_helper.hpp>
+
+#include <hpx/errors/throw_exception.hpp>
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/naming.hpp>
+#include <hpx/include/util.hpp>
+
+#include <array>
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+#include <blaze_tensor/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace dist_matrixops { namespace primitives {
+    ///////////////////////////////////////////////////////////////////////////
+    execution_tree::match_pattern_type const dist_identity::match_data = {
+
+        hpx::util::make_tuple("identity_d", std::vector<std::string>{R"(
+                identity_d(
+                    _1_sz,
+                    __arg(_2_tile_index, nil),
+                    __arg(_3_numtiles, nil),
```

Comment:
As @taless474 has suggested in #1147:
```suggestion
                    __arg(_2_tile_index, find_here()),
                    __arg(_3_numtiles, num_localities()),
```

Reply:
Will do.